### PR TITLE
use mio_serial repo using serialport-rs 2.0.0

### PR DESCRIPTION
**Pull request by ndusart**
Following berkowski/mio-serial#1, it uses the new error type from `mio-serial`.

I removed `mio` dependency as it is not used directly in this crate.

### CODE REVIEW

**Comment by berkowski** [comment]
Once we get berkowski/mio-serial#1 into it's own branch I'd like to do the same here.  You can adjust the dependency target in `Cargo.toml` accordingly to see how things build against the branched `mio-serial`, and we can track upstream fixes until things are ready to merge back into master.

File: Cargo.toml [hunk 1]
```diff
@@ -16,5 +16,4 @@ futures = "0.1"
 tokio-core = "0.1"
 tokio-io = "0.1"
 bytes = "0.4"
-mio = "0.6"
-mio-serial = "0.5"
+mio-serial = { git = "https://github.com/berkowski/mio-serial" }
```
Comment (berkowski) [inline Cargo.toml]: Please change to `mio-serial = { git = "https://github.com/berkowski/mio-serial", branch = "serialport-2"` }

**Comment by berkowski** [comment]
Please switch PR target to the new `serialport-2` branch and update the `mio-serial` dependency to point at the similarly named branch.  Should get some better test results.

**Comment by ndusart** [comment]
Branches are moved to `serialport-2`.

Tests are better, it's just errors from `mio-serial` bubbling up here.

**Comment by berkowski** [comment]
Yup!  Moving forward this week when I've got time.

**Comment by berkowski** [comment]
Merging now that `serialport-2` branch for `mio-serial` has been merged to master.

**Pull request merged**